services/src/error.rs

```diff
@@ -128,6 +128,11 @@ pub enum Error {
 
     TokioPostgresTimeout,
 
```

Comment:
When you start the data atlas, you get

```
error | "IdStringMustBeUuid"
-- | --
message | "IdStringMustBeUuid"
```

This worked in the InMemory-Context.

Reply:
the error is:

> IdStringMustBeUuid { found: "c6f78d3b-609b-454f-8ad5-ed6ec12a9ac0c" }

https://www.freecodeformat.com/validate-uuid-guid.php says it is in fact not a valid uuid. I am not sure why.

The in memory backend didn't check this. So I guess we need to change the dataset-defs and the frontend. Or do you see another solution?

Reply:
Hm, but this site (https://cloud.magiclen.org/us/uuid/checker) says it is valid. Maybe duckduckgo creates uuids that are only valid for certain validators :thinking: 

Can we get more info on that? What does postgres check?

Reply:
It's not postgres, but our Rust code that checks whether the layer/collection id we receive in the API as a string, is im fact a UUID:

```
  #[test]
    fn it_is_a_uuid() {
        uuid::Uuid::from_str("c6f78d3b-609b-454f-8ad5-ed6ec12a9ac0c").unwrap();
    }
```

> thread 'contexts::tests::it_is_a_uuid' panicked at 'called `Result::unwrap()` on an `Err` value: Error(GroupLength { group: 4, len: 13, index: 25 })', services/src/contexts/mod.rs:523:71

Reply:
>  error: SerdeJson: invalid type: map, expected a string consisting of a namespace and name name, separated by a colon, only using alphanumeric characters, underscores & dashes at line 22 column 3

the layer-def in kailin_hatlestad.json still used the old datasetIds. I changed it in the dataset-defs PR

Reply:
you need to merge the main for dataset-defs in order to try this out